genanki/note.py

```diff
@@ -27,12 +28,28 @@ def sort_field(self, val):
   # __init__ and it'll still work.
   @cached_property
   def cards(self):
-    rv = []
-    for card_ord, any_or_all, required_field_ords in self.model._req:
-      op = {'any': any, 'all': all}[any_or_all]
-      if op(self.fields[ord_] for ord_ in required_field_ords):
-        rv.append(Card(card_ord))
-    return rv
+    if self.model.model_type == 0:
+      rv = []
+      for card_ord, any_or_all, required_field_ords in self.model._req:
+        op = {'any': any, 'all': all}[any_or_all]
+        if op(self.fields[ord_] for ord_ in required_field_ords):
+          rv.append(Card(card_ord))
+      return rv
+    # returns a Card with unique ord for each unique cloze reference
+    assert self.model.model_type == 1, self.model.model_type
+    card_ords = set()
+    # find cloze replacements in first template's qfmt, e.g "{{cloze::Text}}"
+    cloze_replacements = set(
+      re.findall(r"{{[^}]*?cloze:(?:[^}]?:)*(.+?)}}", self.model.templates[0]['qfmt']) +
+      re.findall("<%cloze:(.+?)%>", self.model.templates[0]['qfmt']))
+    for field_name in cloze_replacements:
+      field_index = next((i for i, f in enumerate(self.model.fields) if f['name'] == field_name), -1)
+      field_value = self.fields[field_index] if field_index >= 0 else ""
+      # update card_ords with each cloze reference N, e.g. "{{cN::...}}"
+      card_ords.update([int(m)-1 for m in re.findall(r"{{c(\d+)::.+?}}", field_value) if int(m) > 0])
```

Comment:
nit: You can remove the `[]` here.

Reply:
Done: https://github.com/kerrickstaley/genanki/pull/37/commits/0fa99f6fe85c556682057b844e14ac5a95a17636